crates/bevy_render/src/globals.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    extract_resource::ExtractResource,
+    render_resource::{ShaderType, UniformBuffer},
+    renderer::{RenderDevice, RenderQueue},
+    Extract, RenderApp, RenderStage,
+};
+use bevy_app::{App, Plugin};
+use bevy_ecs::prelude::*;
+use bevy_reflect::Reflect;
+use bevy_time::Time;
+
+pub struct GlobalsPlugin;
+
+impl Plugin for GlobalsPlugin {
+    fn build(&self, app: &mut App) {
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app
+                .init_resource::<GlobalsBuffer>()
+                .init_resource::<Time>()
+                .add_system_to_stage(RenderStage::Extract, extract_time)
```

Comment:
This can use the `ExtractResourcePlugin`.

Reply:
That would mean adding a dependency on bevy_render for bevy_time. Is this acceptable?

Reply:
Hmm, probably not as people will want to build without `bevy_render` but use `Time`.

Reply:
Yeah, that's what I was thinking, so I guess we can just keep it like I did and not use the Plugin